extensions/gitpod/src/remoteConnector.ts

```diff
@@ -379,63 +380,149 @@ export default class LocalApp extends Disposable {
 		}
 	}
 
+	private async getWorkspaceSSHDestination(workspaceId: string): Promise<string | undefined> {
+		const session = await vscode.authentication.getSession(
+			'gitpod',
```

Comment:
should not it host specific somehow?

Reply:
it's fine, there's different keychains for different hosts but just one auth provider, keychains are swapped internally

Reply:
I don't think it is cool to reconfigure to another host for one time request.